.github/workflows/pr-ci.yml

```diff
@@ -150,9 +169,68 @@ jobs:
 
           To run smoketest:
           ```
-          # amd64          
+          # amd64
           CRYOSTAT_IMAGE=${{ env.amd64_image }} sh smoketest.sh
 
           # or arm64
           CRYOSTAT_IMAGE=${{ env.arm64_image }} sh smoketest.sh
           ```
+
+  integration-test:
+    needs: [checkout-branch, push-to-ghcr]
+    strategy:
+      matrix:
+        arch: [amd64, arm64]
+    uses: ./.github/workflows/integrated-test.yml
+    with:
+      build-arch: ${{ matrix.arch }}
+      pr-number: ${{ github.event.issue.number }}
+      sha-value: ${{ needs.checkout-branch.outputs.PR_head_sha }}
+      skip-itests: ${{ matrix.arch == 'arm64' }}
+  
+  retest-integration:
+    needs: [checkout-branch]
+    if: contains(github.event.comment.body, '/retest')
+    strategy:
+      matrix:
+        arch: [amd64, arm64]
+    uses: ./.github/workflows/integrated-test.yml
+    with:
+      build-arch: ${{ matrix.arch }}
+      pr-number: ${{ github.event.issue.number }}
+      sha-value: ${{ needs.checkout-branch.outputs.PR_head_sha }}
+      skip-itests: ${{ matrix.arch == 'arm64' }}
+      
+  integration-test-pass:
+    runs-on: ubuntu-latest
+    needs: [integration-test]
+    steps:
+    - name: Leave Actions Run Comment
+      uses: actions/github-script@v6
+      with:
+        script: |
+          const runURL = `https://github.com/${context.repo.owner}/${context.repo.repo}/actions/runs/${{ github.run_id }}`;
+          const commentBody = `both build and itest completed successfully ✅. \n[View Actions Run](${runURL}).`;
```

Comment:
Is there any way to leave a similar comment in the case that the (re)test run fails?

Reply:
I really want this to be part of this PR. I had tried it earlier and still trying and testing it. I tried several ways, with the status itself i.e `success` and status code `$? != 0` still not successful. Looks like the `needs` updates the current tests. The results were different with many times the test being skipped. I'll change the PR status until this is done to be reviewed again.